core/templates/dev/head/services/ExplorationFeaturesService.ts

```diff
@@ -17,44 +17,52 @@
  *               the exploration editor.
  */
 
-var oppia = require('AppInit.ts').module;
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-oppia.factory('ExplorationFeaturesService', [function() {
-  var settings = {
-    areParametersEnabled: false,
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationFeaturesService {
+  static settings = {
     isImprovementsTabEnabled: false,
     isPlaythroughRecordingEnabled: false,
-  };
+    areParametersEnabled: false
+  }
 
-  return {
-    init: function(explorationData, featuresData) {
-      settings.isImprovementsTabEnabled =
-        featuresData.is_improvements_tab_enabled;
-      settings.isPlaythroughRecordingEnabled =
-        featuresData.is_exploration_whitelisted;
-      if (explorationData.param_changes &&
-          explorationData.param_changes.length > 0) {
-        this.enableParameters();
-      } else {
-        for (var state in explorationData.states) {
-          if (explorationData.states[state].param_changes.length > 0) {
-            this.enableParameters();
-            break;
-          }
+  init(explorationData, featuresData) {
+    ExplorationFeaturesService.settings.isImprovementsTabEnabled =
+      featuresData.is_improvements_tab_enabled;
+    ExplorationFeaturesService.settings.isPlaythroughRecordingEnabled =
+      featuresData.is_exploration_whitelisted;
+    if (explorationData.param_changes &&
+        explorationData.param_changes.length > 0) {
+      this.enableParameters();
+    } else {
+      for (var state in explorationData.states) {
+        if (explorationData.states[state].param_changes.length > 0) {
+          this.enableParameters();
+          break;
         }
       }
-    },
-    areParametersEnabled: function() {
-      return settings.areParametersEnabled;
-    },
-    isImprovementsTabEnabled: function() {
-      return settings.isImprovementsTabEnabled;
-    },
-    isPlaythroughRecordingEnabled: function() {
-      return settings.isPlaythroughRecordingEnabled;
-    },
-    enableParameters: function() {
-      settings.areParametersEnabled = true;
-    },
-  };
-}]);
+    }
+  }
+  areParametersEnabled() {
+    return ExplorationFeaturesService.settings.areParametersEnabled;
+  }
+  isImprovementsTabEnabled() {
+    return ExplorationFeaturesService.settings.isImprovementsTabEnabled;
+  }
+  isPlaythroughRecordingEnabled() {
+    return ExplorationFeaturesService.settings.isPlaythroughRecordingEnabled;
+  }
+  enableParameters() {
+    ExplorationFeaturesService.settings.areParametersEnabled = true;
+  }
+}
+
+var oppia = require('AppInit.ts').module;
```

Comment:
Can this be replaced by something like `import { oppia } from 'AppInit.ts';` so we use just these imports in the migrated files?

Reply:
Actually, if you see [here](https://github.com/oppia/oppia/blob/f52947f8628dba82f307a2f3d4f70225a953d20e/core/templates/dev/head/AppInit.ts#L54) in AppInit.ts, the export is a CommonJS style export and such exports can only be imported by CommonJS style imports and `import { oppia } from 'AppInit.ts'` would be an ES6 style import which won't do.
I tried using the ES6 style export for the `var oppia` but I couldn't get it working. Also, it makes sense to distinguish the Angular 8 stuff to be in ES6 style import/export and AngularJS to be in CommonJS style since we already have CommonJS style imports for theAngularJS part of our codebase.

Reply:
Okay